python/llm/src/bigdl/llm/transformers/model.py

```diff
@@ -124,11 +124,26 @@ def from_pretrained(cls,
         :param imatrix: str value, represent filename of importance matrix pretrained on
             specific datasets for use with the improved quantization methods recently
             added to llama.cpp.
+        :param model_hub: str value, options are ``'huggingface'`` and ``'modelscope'``,
+            specify the model hub. Default to be ``'huggingface'``.
         :return: a model instance
         """
         pretrained_model_name_or_path = kwargs.get("pretrained_model_name_or_path", None) \
             if len(args) == 0 else args[0]
-        config_dict, _ = PretrainedConfig.get_config_dict(pretrained_model_name_or_path)
+        model_hub = kwargs.pop("model_hub", "huggingface")
+        invalidInputError(model_hub in ["huggingface", "modelscope"],
+                          "The parameter `model_hub` is supposed to be `huggingface` or "
+                          f"`modelscope`, but got {model_hub}.")
+        if model_hub == "huggingface":
+            config_dict, _ = PretrainedConfig.get_config_dict(pretrained_model_name_or_path)
+        elif model_hub == "modelscope":
+            warnings.warn("The model hub is specified to be modelscope, "
+                          "please make sure you have installed it.")
```

Comment:
why do we need the warning here?

Reply:
> why do we need the warning here?

Users need to install `modelscope` by themselves, so I added the warning previously to remind users. Have removed it.